lib/collection/src/collection_manager/segments_searcher.rs

```diff
@@ -389,44 +390,49 @@ impl SegmentsSearcher {
                 let version = segment.point_version(id).ok_or_else(|| {
                     OperationError::service_error(format!("No version for point {id}"))
                 })?;
-                // If this point was not found yet or this segment have later version
-                if !point_version.contains_key(&id) || point_version[&id] < version {
-                    point_records.insert(
+
+                // If we already have the latest point version, keep that and continue
+                let version_entry = point_version.entry(id);
+                if matches!(version_entry, Entry::Occupied(ref entry) if *entry.get() >= version) {
```

Comment:
Eeeew, `ref`! 🤢

```suggestion
                if matches!(&version_entry, Entry::Occupied(entry) if *entry.get() >= version) {
```

Reply:
Hopefully this works, or I'll look silly 🤡

Reply:
It does